File: frontend/app/types/prompt.ts
// app/types/prompt.ts

export interface PromptType {
    id: number;
    promptType: string;
    description?: string;
    createdAt: string;
    updatedAt: string;
}

export interface Prompt {
    id: number;
    name: string;
    text: string;
    type: PromptType;
    createdAt: string;
    updatedAt: string;
    deletedDate?: string;
    deleted: boolean;
}

export interface PromptCreateRequest {
    name: string;
    text: string;
    typeId: number;
}

export interface PromptUpdateRequest {
    name: string;
    text: string;
    typeId: number;
}

File: backend/src/main/java/net/sampsoftware/genai/model/Prompt.java
package net.sampsoftware.genai.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "prompt")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Prompt {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name", nullable = false, length = 255)
    private String name;

    @Column(name = "text", nullable = false, columnDefinition = "text")
    private String text;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "type_id", nullable = false, foreignKey = @ForeignKey(name = "prompt_type_id_fkey"))
    private PromptType type;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "deleted_date")
    private LocalDateTime deletedDate;

    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public boolean isDeleted() {
        return deletedDate != null;
    }

    public void markDeleted() {
        this.deletedDate = LocalDateTime.now();
    }
}